### ISSUE TITLE
[FR] option to users to be able to edit detection-rules

### ISSUE THREAD

**Issue by krishhteja** [elastic/detection-rules#1917]
**Is your feature request related to a problem? Please describe.**
1) For a project, I'm trying to add detection rules such that they are tied to a particular namespace. Due to the parsing of beats using `beat-*`, it gets harder to add a namespace variable before *
2) If any of the parent directories have a `-` in the name, it fails to compile

**Describe the solution you'd like**
1) Once the issue is resolved, beats would be parsed by `beat-` instead of `beat-*` so that it is easy to add a namespace after `beat-`
2) Once fixed, parent directories having `-` shouldn't block from running the program

**Describe alternatives you've considered**
2) Edited my file path to have no -. This works fine but having a generic option would help

**Additional context**

**Comment by brokensound77** [comment]
Can you share an example similar to an index pattern you are using with namespace

**Comment by krishhteja** [comment]
Sure. I'm trying to add rules having namespace in them `logbeat-<NAMESPACE>*`, `winbeat-<NAMESPACE>*` etc so that my search for that rule is limited to only that namespace

**Comment by krishhteja** [comment]
This is the [PR](https://github.com/elastic/detection-rules/pull/1856) for the same

**Comment by jameswiggins** [comment]
@krishhteja, Could you please elaborate more on your work around for this? 

I also need to add a namespace to the indices that these rules query. I am utilizing spaces in Kibana and each space is configured to show data with a specific [namespace in the index naming scheme](https://www.elastic.co/guide/en/fleet/8.9/data-streams.html#data-streams-naming-scheme). I would like to be able to deploy these detection rules to each space, and have the rules in each space only query the indices with a given value in the namespace. e.g. the winlogbeat-* index pattern is too generic for me. 

For example, I have a Kibana space called Production. I would like to deploy all of the detection rules to my Production space and have them only query the winlogbeat-production-* indices. How do I accomplish that?

**Comment by krishhteja** [comment]
We took a different approach for this problem. 

We used generic rules which would query existing indices. Our change was on pipelines - for all the pipelines, we added an enricher which would add details of space or an identifier which would differentiate groups. Once this is done, every entry in the indices would have enriched data. We have a cron trigger which would reindex main index to space level index based on the identifier 

example: 
```
source: {
      index: '.alerts-security.alerts-default',
      query: {
        bool: {
          must: [
            {
              match: {
                'IDENTIFIER': groupID,
              },
            },
            {
              range: {
                '@timestamp': {
                  gte: moment().utc().subtract(minutesToSubtract, 'minutes').format('YYYY-MM-DDTHH:mm:ss'),
                },
              },
            },
          ],
          should: [],
        },
      },
    },
    dest: {
      index: `.alerts-security.alerts-${getElasticNamespace(groupID)}`,
    },
```

**Comment by jameswiggins** [comment]
That is great @krishhteja, thank you for sharing! I will do the same since I already have the identifier field in each document. 

How did you bootstrap the space level index? Did you have to duplicate/modify `.alerts-security.alerts-default-index-template`?

I suppose with this method you lose the ability to have different exclusions in each space?

**Comment by jameswiggins** [comment]
Having thought a little more about it, there are some concerns with doing it that way. 1) this method does not work for aggregation/threshold based alerts, since they would be querying the combined data over all namespaces. 2) you lose the ability to have namespace specific exceptions.

**Comment by Mikaayenson** [comment]
@krishhteja circling back to this issue. Did you run into issues with unit tests? It's hard to tell from the [PR](https://github.com/elastic/detection-rules/pull/1856/files). Perhaps you could share an exported rule NDJSON from Kibana and the error you ran into?

**Comment by eric-forte-elastic** [comment]
With this https://github.com/elastic/detection-rules/pull/3889 new feature, we now support providing custom schemas which should resolve this issue. If you still are running into any problems, feel free to re-open or create a new issue.
